packages/babel-traverse/src/path/family.ts

```diff
@@ -109,41 +122,157 @@ function completionRecordForSwitch(cases, paths) {
   return paths;
 }
 
-function _getCompletionRecords(path: NodePath): Completion[] {
-  let records = [];
+function completionRecordForSwitch(
+  cases: NodePath<t.SwitchCase>[],
+  records: Completion[],
+  context: CompletionContext,
+): Completion[] {
+  // https://tc39.es/ecma262/#sec-runtime-semantics-caseblockevaluation
+  let lastNormalCompletions = [];
+  for (let i = 0; i < cases.length; i++) {
+    const casePath = cases[i];
+    const caseCompletions = _getCompletionRecords(casePath, context);
+    const normalCompletions = [];
+    const breakCompletions = [];
+    for (const c of caseCompletions) {
+      if (c.type === NORMAL_COMPLETION) {
+        normalCompletions.push(c);
+      }
+      if (c.type === BREAK_COMPLETION) {
+        breakCompletions.push(c);
+      }
+    }
+    if (normalCompletions.length) {
+      lastNormalCompletions = normalCompletions;
+    }
+    records = records.concat(breakCompletions);
+  }
+  records = records.concat(lastNormalCompletions);
+  return records;
+}
+
+function markCompletionsAs(completions: Completion[], type) {
+  completions.forEach(c => {
+    c.type = type;
+  });
+}
 
+function getStatementListCompletion(
+  paths: NodePath[],
+  context: CompletionContext,
+): Completion[] {
+  let completions = [];
+  if (context.canHaveBreak) {
+    let lastNormalCompletions = [];
+    for (let i = 0; i < paths.length; i++) {
+      const path = paths[i];
+      const newContext = { ...context, inCaseClause: false };
+      if (
+        path.isBlockStatement() &&
+        (context.inCaseClause || context.shouldPopulateBreak)
+      ) {
+        newContext.shouldPopulateBreak = true;
+      } else {
+        newContext.shouldPopulateBreak = false;
+      }
+      const statementCompletions = _getCompletionRecords(path, newContext);
+      if (
+        statementCompletions.length > 0 &&
+        statementCompletions.every(c => c.type === BREAK_COMPLETION)
+      ) {
+        if (
+          lastNormalCompletions.length > 0 &&
+          statementCompletions.every(c =>
+            c.path.isBreakStatement({ label: null }),
+          )
+        ) {
+          markCompletionsAs(lastNormalCompletions, BREAK_COMPLETION);
+          completions = completions.concat(lastNormalCompletions);
+          statementCompletions.forEach(c => c.path.remove());
```

Comment:
I know that this was already the previous behavior, but we should add a comment explaining that this function mutates the AST. I didn't expect it, given the function name.